custom_model_runner/datarobot_drum/drum/gpu_predictors/nemo_predictor.py

```diff
@@ -0,0 +1,250 @@
+        # column_names = ["row_id", "choice_id", "completions"]
+        column_names = ["completions"]
+
+        return RawPredictResponse(np.array(results), np.array(column_names))
+
+    def _get(self, row, column_name):
+        try:
+            return row[column_name]
+        except KeyError:
+            expected_column_names = [self.user_prompt_column]
+            if self.use_chat_context:
+                expected_column_names.append(self.assistant_response_column)
+            raise DrumCommonException(f"Model expects column names '{expected_column_names}'")
+
+    def _create_completions(self, messages, row_id=0):
+        if self.system_prompt_value:
+            # only the first chat message can have the system role
+            messages.insert(0, {"role": ChatRoles.SYSTEM, "content": self.system_prompt_value})
+
+        try:
+            completions = self.nim_client.chat.completions.create(
+                model=self.model_name,
+                messages=messages,
+                n=self.num_choices_per_completion,
+                temperature=self.temperature,
+                max_tokens=self.max_tokens,
+            )
+        except openai.BadRequestError:
+            self.logger.error("Payload: %s", json.dumps(messages), exc_info=True)
+            raise DrumCommonException("Bad payload")
+
+        completion_choices = [
+            # [row_id, choice_id, choice.message.content]
+            choice.message.content
+            for choice_id, choice in enumerate(completions.choices)
+        ]
+
+        self.logger.debug("results: %s", completion_choices)
+        return completion_choices
+
+    def predict_unstructured(self, data, **kwargs):
+        raise DrumCommonException("The unstructured target type is not supported")
+
+    def _transform(self, **kwargs):
+        raise DrumCommonException("Transform feature is not supported")
+
+    def _run_nemo_server(self, nemo_process: DrumServerProcess):
+        cmd = [
+            "nemollm_inference_ms",
+            "--model",
+            self.model_name,
+            "--health_port",
+            self.health_port,
+            "--openai_port",
+            self.openai_port,
+            "--nemo_port",
+            self.nemo_port,
+            "--num_gpus",
+            self.gpu_count,
+            "--log_level",
+            "info",
+        ]
+
+        # update the path so nemollm process can find its libraries
+        env = os.environ.copy()
+        datarobot_venv_path = os.environ.get("VIRTUAL_ENV")
+        env["PATH"] = env["PATH"].replace(f"{datarobot_venv_path}/bin:", "")
+        with subprocess.Popen(
+            cmd,
+            env=env,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            preexec_fn=os.setsid,
+        ) as p:
+            nemo_process.process = p
+            for line in p.stdout:
+                self.logger.info(line[:-1])
+
```

Comment:
Is this necessary/preferred? Does the output not show up well if we just let Triton log to the normal stderr/out?